pwndbg/aglib/memory.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import annotations
+
+from typing import Dict
+from typing import Set
+from typing import Union
+
+import pwndbg.aglib.arch
+import pwndbg.aglib.typeinfo
+import pwndbg.lib.cache
+import pwndbg.lib.memory
+from pwndbg.lib.memory import PAGE_SIZE
+
+GdbDict = Dict[str, Union["GdbDict", int]]
+
+
+MMAP_MIN_ADDR = 0x8000
+
+
+def read(addr: int, count: int, partial: bool = False) -> bytearray:
```

Comment:
This adds an unnecessary call atm with no benefit.

Ideally this should just be `read = gdb_or_lldb_read_memory_func` since their API is the same anyway?

Reply:
It would probably have to look something like this, but it would work:

```py
read: Callable[..., bytearray] = None

@pwndbg.dbg.event_handler(EventType.START)
def on_start():
    global read
    read = pwndbg.dbg.selected_inferior().read_memory

@pwndbg.dbg.event_handler(EventType.EXIT)
def on_exit():
    global read
    read = None
```